server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -340,6 +350,24 @@ public ScheduledCancellable schedule(Runnable command, TimeValue delay, String e
         command = threadContext.preserveContext(command);
         if (Names.SAME.equals(executor) == false) {
             command = new ThreadedRunnable(command, executor(executor));
+        } else if (slowSchedulerWarnThresholdNanos > 0) {
+            final Runnable finalCommand = command;
```

Comment:
Let's make all the `Runnable` things in scope `final` rather than mutating `command`.

Reply:
++